common/web/gesture-recognizer/src/engine/headless/gestures/specs/contactModel.ts

```diff
@@ -0,0 +1,42 @@
+import { PathModel } from "./pathModel.js";
+
+// For modipress, in particular - says what gestures are allowed during the 'push' state.
+export interface PushResult {
+  type: 'push';
+  permittedGestures: string[]; // of ids for allowed types
+}
+
+export interface ComplexResolveResult {
+  type: 'resolve';
+  cancelBaseAwait?: boolean; // default false, but if true, will auto-cancel the top-level model's await (if it exists)
+}
+
+type WrappedString<Text> = { type: Text };
+
+// pop - ends the modipress 'push' state.
+type SimpleStringResult = 'resolve' | 'reject' | 'pop';
+
+export type PointModelResolution =  PushResult | ComplexResolveResult | WrappedString<SimpleStringResult>;
+
+
+export type SpecPointModelResolution = SimpleStringResult | PointModelResolution;
+
+export interface ContactModel {
+  pathModel: PathModel, // Might be able to fully drop the generic.
+  onPathResolve: SpecPointModelResolution,
```

Comment:
The prefix `on` is usually used for event listeners. Perhaps a different name? `pathResolutionAction`?